jenkins/bootstrap.py

```diff
@@ -385,10 +385,17 @@ def metadata(repo, artifacts):
     if not meta or not isinstance(meta, dict):
         meta = {}
     meta['repo'] = repo
+
+    try:
+        commit = call(['git', 'rev-parse', '--short=14', 'HEAD^{commit}'], output=True)
```

Comment:
I cargo-culted this from elsewhere, but it probably doesn't matter for something like `HEAD`. It is more explicit about resolving to return a commit sha1, however.

From the [docs](https://www.kernel.org/pub/software/scm/git/docs/gitrevisions.html#_specifying_revisions):
> <rev>^{<type>}, e.g. v0.99.8^{commit}
A suffix ^ followed by an object type name enclosed in brace pair means dereference the object at <rev> recursively until an object of type <type> is found or the object cannot be dereferenced anymore (in which case, barf). For example, if <rev> is a commit-ish, <rev>^{commit} describes the corresponding commit object. Similarly, if <rev> is a tree-ish, <rev>^{tree} describes the corresponding tree object. <rev>^0 is a short-hand for <rev>^{commit}.

It looks like it matters more if we were dereferencing a tag. e.g. on k/k:
```console
$ git rev-parse v1.5.3
2d49a2edd9e387873da2e5b5dbe74d61c9b95665
$ git rev-parse v1.5.3^{commit}
029c3a408176b55c30846f0faedf56aae5992e9
$ git rev-parse v1.5.3^{object}
2d49a2edd9e387873da2e5b5dbe74d61c9b95665
```

I think 2d49... is referring to the tag object, whereas 029c... is referring to the actual commit pointed to by the tag `v1.5.3`.